### PR TITLE
add first plugin archetype

### LINKED ISSUE #1: first plugin archetype
first plugin archetype

### PR DESCRIPTION
**Pull request by daizhenyu**
**What type of PR is this?**

Archetype.

**What this PR does / why we need it?**

add first plugin archetype

**Which issue(s) this PR fixes？**

Fixes #1

### CODE REVIEW

File: archetype.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+archetype.groupId=io.sermant
+archetype.artifactId=sermant-template-archetype
+archetype.version=2.1.0
+package=io.sermant
+excludePatterns=.github/**,.idea/**,.idea/libraries/*,**/*.iml,README.md,LICENSE
```
Comment (lilai23) [inline archetype.properties]: move LICENSE

File: template/config/config.yaml
Comment (lilai23) [inline template/config/config.yaml:1]: add comment for config development

File: template/template-plugin/src/main/java/io/sermant/template/EchoService.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package io.sermant.template;
```
Comment (lilai23) [inline template/template-plugin/src/main/java/io/sermant/template/EchoService.java:17]: copyright

File: template/template-service/src/main/java/io/sermant/template/EchoServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * @author luanwenfei
+ * @since 2023-02-15
+ */
+public class EchoServiceImpl implements EchoService {
+    @Override
+    public void echo(String string) {
+        // respond to a greeting
+        string = string.replaceAll("[\\pP+~$`^=|<>～｀＄＾＋＝｜＜＞￥×]", "");
+        System.out.println("ECHO: " + string + " to you!");
+    }
```
Comment (lilai23) [inline template/template-service/src/main/java/io/sermant/template/EchoServiceImpl.java:31]: Start and Stop method

File: template/template-plugin/src/main/java/io/sermant/template/TemplateDeclarer.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+public class TemplateDeclarer extends AbstractPluginDeclarer {
+    @Override
+    public ClassMatcher getClassMatcher() {
+        return ClassMatcher.nameEquals("io.sermant.demo.template.Application");
+    }
+
+    @Override
+    public InterceptDeclarer[] getInterceptDeclarers(ClassLoader classLoader) {
+        return new InterceptDeclarer[]{
+                InterceptDeclarer.build(MethodMatcher.nameEquals("greeting"), new Interceptor() {
```
Comment (lilai23) [inline template/template-plugin/src/main/java/io/sermant/template/TemplateDeclarer.java]: put interceptor in a new file

File: archetype.properties (hunk 1 above)
Comment (daizhenyu) [reply archetype.properties]: done

File: template/config/config.yaml
Comment (daizhenyu) [reply template/config/config.yaml:1]: done

File: template/template-plugin/src/main/java/io/sermant/template/EchoService.java (hunk 2 above)
Comment (daizhenyu) [reply template/template-plugin/src/main/java/io/sermant/template/EchoService.java:17]: done

File: template/template-service/src/main/java/io/sermant/template/EchoServiceImpl.java (hunk 3 above)
Comment (daizhenyu) [reply template/template-service/src/main/java/io/sermant/template/EchoServiceImpl.java:31]: done

File: template/template-plugin/src/main/java/io/sermant/template/TemplateDeclarer.java (hunk 4 above)
Comment (daizhenyu) [reply template/template-plugin/src/main/java/io/sermant/template/TemplateDeclarer.java]: done

File: .github/workflows/create_release.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          echo "version=${str##*v}" >> $GITHUB_ENV
+      - name: Checkout Source
+        uses: actions/checkout@v4
+      - name: Set up JDK 8
+        uses: actions/setup-java@v4
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+          cache: maven
+      - name: Build with Maven
```
Comment (hanbingleixue) [inline .github/workflows/create_release.yml:32]: The license file is not generated.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,36 @@
-# first-plugin
-Demo plugin of Sermant
+## What is it?
```
Comment (hanbingleixue) [inline README.md:6]: It is recommended to add a Chinese README file.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,36 @@
...
+The quick development and usage guide for implementing service governance plugins under the Sermant framework can be found at https://bbs.huaweicloud.com/blogs/415083.
+
+## How to deploy first-plugin archetype using github action?
+### 1. Modify archetype version
+Modify archetype.properties file:
+```properties
+archetype.groupId=com.huaweicloud.sermant
+archetype.artifactId=sermant-template-archetype
+archetype.version=2.1.0 # modify this version before deploy
+excludePatterns=.github/**,.idea/**,.idea/libraries/*,**/*.iml,README.md,LICENSE
```
Comment (hanbingleixue) [inline README.md]: Modify the archetype.properties file to be the same as the latest archetype.properties file.

File: application/pom.xml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+            <version>2.7.15</version>
```
Comment (hanbingleixue) [inline application/pom.xml]: The version number must be defined in properties.

File: application/pom.xml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-maven-plugin</artifactId>
+                <version>2.7.15</version>
```
Comment (hanbingleixue) [inline application/pom.xml]: Same as above

File: README.md (hunk 7 above)
Comment (daizhenyu) [reply README.md]: done

File: .github/workflows/create_release.yml (hunk 5 above)
Comment (daizhenyu) [reply .github/workflows/create_release.yml:32]: down

File: application/pom.xml (hunk 8 above)
Comment (daizhenyu) [reply application/pom.xml]: down

File: application/pom.xml (hunk 9 above)
Comment (daizhenyu) [reply application/pom.xml]: down

File: README.md (hunk 6 above)
Comment (daizhenyu) [reply README.md:6]: done

**Review by hanbingleixue — APPROVED** [review]

**Review by lilai23 — APPROVED** [review]

**Pull request merged**